source/Week_11/Graph avec des classes.ipynb

```diff
@@ -0,0 +1,121 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Créer des graphes avec des classes\n",
+    "\n",
+    "Les semaines précédentes, vous avez vu comment conceptualiser des graphes avec des dictionnaires et des listes. Cette fois nous allons travailler avec des classes pour représenter des graphes orientés."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercice 1\n",
+    "\n",
+    "Pour conceptualiser notre graphes nous allons utliser deux classes diffenrentes, les sommets seront des chaines de caratères, indivuelement `vertex`;\n",
+    "+ une classe `Edge` pour definir les arretes.\n",
+    "Chaque instance aura comme attribut un sommet de départ `from_vertex`, un sommet d'arrivée `to_vertex` et un poids `weight`. La classe aura comme unique methode `__repr__`pour retourner l'objet sous la forme d'un dictionaire.\n",
+    "+ une classe `Graph`pour definir un graphe.\n",
+    "Chaque instance aura comme attribut un set `edges` contenant les arêtes (qui seront des objets) et un set `vertices` contenant les sommets. astuce: pour initialiser un set vide utilisez `set()`\n",
+    "\n",
+    "\n",
+    "La classe `Graph` aura comme méthodes:\n",
+    "+ `add_vertex` pour créer un sommet\n",
+    "+ `edge_exist` pour vérifier si une arête existe, elle retourne `weight` si oui, `False` sinon. \n",
+    "+ `update_weight` pour mettre le poids à jour d'une connexion existante, et appeler la méthode `new_edge` sinon.\n",
+    "+ `new_edge` pour créer une instance de `Edge`et l'ajouter au set `edges` si la connexion n'existe pas déjà, si elle existe avec un autre poids mettre à jour le poids, si elle existe de façon identique alors retournez le dans la console avec `print(...)`. Astuce: Ici il peut être judicieux de créer une méthode privée `__generate_edge` et de l'appeler puisque avant de créer une connexion il faut s'assurer que les sommets existes et les créer sinon. \n",
+    "+ `del_edge` pour supprimer un `edge`\n",
+    "+ `__repr__` pour retourner l'objet sous forme de dictionnaire. \n",
```

Comment:
__repr__ is meant to returns a string. If you do `repr(x)` and it returns a dict, you are going to have an error

Reply:
ok thank you for to review, that's why I use return str({...}) in the solution. I usually do this to a have a factory method from_dict() for debugging purposes (this is more a hack than standars/good practice I assume)

Reply:
I didn't see before, my bad!